pinot-common/src/main/java/org/apache/pinot/common/response/broker/BrokerResponseNativeV2.java

```diff
@@ -83,6 +84,16 @@ public ResultTable getResultTable() {
   @Override
   public void setResultTable(@Nullable ResultTable resultTable) {
     _resultTable = resultTable;
+    // NOTE: Update _numRowsResultSet when setting non-null result table. We might set null result table when user wants
+    //       to hide the result but only show the stats, in which case we should not update _numRowsResultSet.
+    if (resultTable != null) {
+      _numRowsResultSet = resultTable.getRows().size();
+    }
+  }
+
+  @Override
+  public int getNumRowsResultSet() {
+    return _numRowsResultSet;
```

Comment:
add setNumRowsResultSet() as well?

Reply:
I didn't add it intentionally because we want to always set it along with the result table

Reply:
I added it for V1 response for JSON ser/de purpose, which is not supported in V2 response as of now